### ISSUE TITLE
Ability to set a custom count method

### ISSUE THREAD

**Issue by matthewrudy** [activeadmin/activeadmin#4182]
For our app we have certain tables which are so big calling `MyModel.count` takes over a second.
And conceivably using some combination of search params could cause a table scan, and possibly crash the database.

So we need a way to override the `total_count` method from Kaminari.
Our current approach is to override the count globally

``` ruby
module Kaminari::ActiveRecordRelationMethods
  def total_count(column_name = :all, options = {})
    @total_count ||= begin
       proprietary_count_method
    end
  end
end
```

This works, but it's a hack, and being global in nature, we have no way to selectively opt-in/out.

So I'm wondering, what's the best way to avoid calling `total_count`?

Options:

A. Enable custom pagination methods

``` ruby
ActiveAdmin.config.pagination_method_name ||= Kaminari.config.page_method_name
```

B. Introduce custom counting methods

``` ruby
ActiveAdmin.config.count_method_name ||= :total_count
```

C. Something else?

**Comment by matthewrudy** [comment]
Looking more into this, I saw this Issue #4133.

So I think we just need a good way to override all counts.

**Comment by timoschilling** [comment]
closed as double of #4133
